tutorials/misc/binary_serialization_api.rst

```diff
@@ -400,19 +404,41 @@ The array data is padded to 4 bytes.
 | 8..8+length\*4   | 4     | Integer   | 32 Bits Signed Integer    |
 +------------------+-------+-----------+---------------------------+
 
-22: :ref:`PoolRealArray<class_poolrealarray>`
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+22: :ref:`PackedInt64Array<class_PackedInt64Array>`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 +------------------+-------+-----------+---------------------------+
 | Offset           | Len   | Type      | Description               |
 +==================+=======+===========+===========================+
-| 4                | 4     |Integer    | Array Length (Floats)     |
+| 4                | 8     | Integer   | Array Length (Integers)   |
 +------------------+-------+-----------+---------------------------+
-| 8..8+length\*4   | 4     |Integer    | 32 Bits IEE 754 Float     |
+| 8..8+length\*8   | 8     | Integer   | 64 Bits Signed Integer    |
```

Comment:
I think the "Array Length" part is always 4.

Also, shouldn't they all be offset by 4 due to this? If so, the existing documentation is wrong.

Reply:
@aaronfranke Well, it would make sense for it to always be 4, but it uses `encode_uint64` for `PackedInt64Array` specifically. It may be an error in `marshalls.cpp`, though. Int32, Float32 and Float64 all use `encode_uint32` for the length.